src/main/scala/strawman/collection/IterableFactories.scala

```diff
@@ -0,0 +1,26 @@
+package strawman
+package collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Int
+
+/** Base trait for instances that can construct a collection from an iterable */
+trait FromIterable[+C[X] <: Iterable[X]] {
+  def fromIterable[B](it: Iterable[B]): C[B]
+}
+
+/** Base trait for companion objects of collections */
+trait IterableFactories[+C[X] <: Iterable[X]] extends FromIterable[C] {
```

Comment:
I thought it would make more sense because this trait provides several factory methods. We can just settle on either `XxxFactory` or `XxxFactories`…